### PR TITLE
update README.md

### PR DESCRIPTION
**Pull request by Sherylueen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,201 @@
...
-- Updating SUPPORT.MD with content about this project's support experience
-- Understanding the security reporting process in SECURITY.MD
-- Remove this section from the README
+The definition of this Github Action is in [action.yml]()
 
-## Contributing
+
+# Pre-requisites:
+* Azure Login Action: Authenticate using [Azure Login](https://github.com/Azure/login)  action. The Get Quick Assessments action assumes that Azure Login is done using an Azure service principal that has [sufficient permissions](https://docs.microsoft.com/en-us/azure/governance/policy/overview#rbac-permissions-in-azure-policy) trigger and get quick assessment on the selected scopes. Once login is done, the next set of actions in the workflow can perform tasks such as geting quick assessments by report or by deployment. For more details on permissions, checkout 'Configure credentials for Azure login action' section in this page  or alternatively you can refer the full [documentation](https://github.com/Azure/login) of Azure Login Action.
+* Create an ACAT report(optional): Go to Azure Portal to create an ACAT report for you application.
```
Comment (HuiquanJiang-ms) [inline README.md]: Looks like report and deployment ID are both optional. What if the user passes none of them?

File: README.md [hunk 2]
```diff
@@ -1,16 +1,201 @@
...
+* Prepare the deployment id(optional): You can also get quick assessment by your deployment, set the deployment id as output in your former deploy action, and take the deployment id as input of Get Quick Assessments action.
+
+
+
+# Inputs for the Action
+
+* `cred`: mandatory. The credential you use to get quick assessments.
```
Comment (HuiquanJiang-ms) [inline README.md]: A question from user's perspective, why do I need this cred when I've already configured the az login action?

File: README.md [hunk 3]
```diff
@@ -1,16 +1,201 @@
...
+* `report-name`: Optional. If you want to get quick assessments by report, you should create a report before you run the github action and set the report-name value the name of the report you created. At least one of the 2 parameters `report-name` and `deployment-id` must be filled.
+* `deployment-id`: Optional. If you want to get quick assessments by deployment, you should get the id of your deployment, and pass the value to `deployment-id`. At least one of the 2 parameters `report-name` and `deployment-id` must be filled.
```
Comment (HuiquanJiang-ms) [inline README.md]: The action will update the report if user passes both `report-name` and `deployment-id`, is that correct?

File: README.md [hunk 4]
```diff
@@ -1,16 +1,201 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Azure login
+      uses: Azure/login@v1.4.6
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: analyse Microsoft 365 compliance results
+      uses: ./ # Uses an action in the root directory
```
Comment (HuiquanJiang-ms) [inline README.md]: This may not work in users' repo

File: README.md (hunk 1 above)
Comment (Sherylueen) [reply README.md]: action will fail directly.

File: README.md [hunk 5]
```diff
@@ -1,16 +1,201 @@
-# Project
+# Get Quick Assessments Action
 
-> This repo has been populated by an initial template to help get you started. Please
-> make sure to update the content to build a great experience for community-building.
+Get Quick Assessments Action is part of App Compliance Automation Tool(ACAT). With Get Quick Assessments Action you can now get quick assessments of Azure Resources by report or by deployment from your Github Workflows. Since the get quick assessments action can be added directly after the resource deployment, you can have a immediately result of the compliance status of resources deployed this time. Its now even easier to follow safe deployment practices and catch non-compliant issues.
```
Comment (qinzhouxu) [inline README.md]: "for Microsoft 365" should not be omitted for here and action name, and all "M365" should be "Microsoft 365"

File: README.md [hunk 6]
```diff
@@ -1,16 +1,201 @@
...
+        resourceGroupName: ${{ secrets.RESOURCE_GROUP}}
+        template: ./deploy/Storage.template.json
+        parameters: storageAccountType=Standard_LRS
+    - run: echo ${{ steps.deployarm.outputs.deploymentId }}
+
+    - name: analyse Microsoft 365 compliance results
+      uses: ./ # Uses an action in the root directory
+      with:
+        cred: ${{ secrets.AZURE_CREDENTIALS }}
+        deployment-name: ${{ steps.deployarm.outputs.deploymentId }}
```
Comment (HuiquanJiang-ms) [inline README.md]: deployment-id?

File: README.md [hunk 7]
```diff
@@ -1,16 +1,201 @@
...
+        
+```
+The above workflow will get quick assessments by deployment. 
+
+
+
+
+
+
+# Configure credentials for Azure login action:
```
Comment (HuiquanJiang-ms) [inline README.md:143]: Shall we place this section in `prerequisites`? BTW I remember this part is covered by Github doc, maybe we can just leave a link here.

File: README.md [hunk 8]
```diff
@@ -1,16 +1,201 @@
...
-- Improving this README.MD file to provide a great experience
-- Updating SUPPORT.MD with content about this project's support experience
-- Understanding the security reporting process in SECURITY.MD
-- Remove this section from the README
+The definition of this Github Action is in [action.yml]()
 
-## Contributing
+
+# Pre-requisites:
+* Azure Login Action: Authenticate using [Azure Login](https://github.com/Azure/login)  action. The Get Quick Assessments action assumes that Azure Login is done using an Azure service principal that has [sufficient permissions](https://docs.microsoft.com/en-us/azure/governance/policy/overview#rbac-permissions-in-azure-policy) trigger and get quick assessment on the selected scopes. Once login is done, the next set of actions in the workflow can perform tasks such as geting quick assessments by report or by deployment. For more details on permissions, checkout 'Configure credentials for Azure login action' section in this page  or alternatively you can refer the full [documentation](https://github.com/Azure/login) of Azure Login Action.
```
Comment (qinzhouxu) [inline README.md]: Should we emphasize detailed permissions here since we will do onboard?

File: README.md (hunk 2 above)
Comment (qinzhouxu) [reply README.md]: Should also emphasize cred is same as login action

File: README.md [hunk 9]
```diff
@@ -1,16 +1,201 @@
...
+* Create an ACAT report(optional): Go to Azure Portal to create an ACAT report for you application.
+* Prepare the deployment id(optional): You can also get quick assessment by your deployment, set the deployment id as output in your former deploy action, and take the deployment id as input of Get Quick Assessments action.
+
+
+
+# Inputs for the Action
+
+* `cred`: mandatory. The credential you use to get quick assessments.
+* `report-name`: Optional. If you want to get quick assessments by report, you should create a report before you run the github action and set the report-name value the name of the report you created. At least one of the 2 parameters `report-name` and `deployment-id` must be filled.
```
Comment (qinzhouxu) [inline README.md]: Maybe add this link here? Not sure whether we have short link for it.

https://learn.microsoft.com/en-us/microsoft-365-app-certification/docs/automate-certification-with-acat

File: README.md [hunk 10]
```diff
@@ -1,16 +1,201 @@
...
+    name: Test artifact
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Azure login
+      uses: Azure/login@v1.4.6
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    - name: Deploy with ARM template
```
Comment (qinzhouxu) [inline README.md:79]: We should also mention Bicep scenario

File: README.md (hunk 1 above)
Comment (Sherylueen) [reply README.md]: Added some description here, please review.

File: README.md [hunk 11]
```diff
@@ -1,16 +1,201 @@
...
+The above workflow will get quick assessments by deployment. 
+
+
+
+
+
+
+# Configure credentials for Azure login action:
+
+With the Azure login Action, you can perform an Azure login using [Azure service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals). The credentials of Azure Service Principal can be added as [secrets](https://help.github.com/en/articles/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables) in the GitHub repository and then used in the workflow. Follow the below steps to generate credentials and store in github.
```
Comment (qinzhouxu) [inline README.md:145]: Is it possible directly link to login action's tutorial instead of duplicate one?

File: README.md (hunk 3 above)
Comment (Sherylueen) [reply README.md]: Yes, description added.

File: README.md (hunk 6 above)
Comment (Sherylueen) [reply README.md]: The deploymentId is defined in the output of last step.
there is another error, which is the parameter name deployment-name, actually we need deployment-id.

File: README.md (hunk 4 above)
Comment (Sherylueen) [reply README.md]: Good point! resolved!

File: README.md (hunk 5 above)
Comment (Sherylueen) [reply README.md]: Please review.

Comment (Sherylueen) [reply README.md]: Good  point， resolved.

File: README.md (hunk 2 above)
Comment (Sherylueen) [reply README.md]: emphasized the cred is the same as login. Do we need extra discussion to decide whether we take cred as input or tenant-id?

File: README.md (hunk 10 above)
Comment (Sherylueen) [reply README.md:79]: Added.

File: README.md (hunk 9 above)
Comment (Sherylueen) [reply README.md]: Added.

**Review by qinzhouxu — APPROVED** [review]

**Pull request merged**